### PR TITLE
Implement new node removal strategy

### PR DESCRIPTION
**Pull request by wallin**
Syncs the node removal strategy with the one from the Java library:
https://github.com/aerospike/aerospike-client-java/blob/7352c15c04015f4bd4f6985e138fa931179fb5d7/client/src/com/aerospike/client/cluster/Cluster.java#L521

### CODE REVIEW

**Comment by wallin** [comment]
@jhecking Other than unit tests, I'm not sure how to best test this. Let me know if you have any ideas

File: lib/aerospike/cluster/find_nodes_to_remove.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+
+module Aerospike
+  class Cluster
+    # Calculates which nodes that should be removed from the cluster
+    module FindNodesToRemove
+      class << self
+        def call(cluster, refresh_count)
```
Comment (jhecking) [inline lib/aerospike/cluster/find_nodes_to_remove.rb]: It looks like we don't actually need the refresh counter - the function just cares whether the counter is zero or not, i.e. whether any node was refreshed successfully or not. So maybe we should just pass `peers.refresh_count.zero?` as a boolean instead? But I'm also ok to keep it as it is - as the algorithm to "find nodes to remove" evolves, we might need the actual counter again in the future. (And also to not unnecessarily diverge from the Java implementation.)

Comment (wallin) [reply lib/aerospike/cluster/find_nodes_to_remove.rb]: Ok, I'll leave it as is for now then, for consistency

**Comment by wallin** [comment]
> But I think you need to reset the Peers#refresh_count counter in Cluster#refresh_nodes inside the peers.generation_changed? branch (line 269).

Good catch. Fixed it

**Pull request merged**